### ISSUE TITLE
UI is unresponsive for a couple seconds when the window becomes focused

### ISSUE THREAD

**Issue by james-rabbit** [ForkIssues/Tracker#94]
When I put the Fork window into focus, it takes a few seconds before it is responsive and I can click on anything or type.

I thought this was because I am working on a "big" repo, but I do not have this issue when I open up the [linux repo](https://github.com/torvalds/linux) using Fork. 

The biggest differences I can see between my repo and the linux one is that mine has hundreds of remote branches, tags and submodules, so I think there may be a slowdown in the Fork app because of that.

In other apps I disable auto-fetching submodules, but Fork does not have an option for that. I am not sure if it is checking submodules every time I focus the window, but that would definitely slow it down if it is.

Thanks!

**Comment by DanPristupov** [comment]
@james-rabbit thank you for reporting that. There should be some performance improvements in this area in the upcoming Fork update (1.0.56). Please wait for it and check if things got better.

As for now, could you please run in command line the following commands and tell me the result?
You can send it to my email (support@git-fork.com) if this is a sensitive information.

Number of branches:
```for-each-ref | wc -l```

Number of local branches:
```for-each-ref refs/heads | wc -l```

Number of Remote branches:
```for-each-ref refs/remotes | wc -l```

Number of tags:
```for-each-ref refs/tags | wc -l```

**Comment by james-rabbit** [comment]
Sure thing,

Number of branches:
```
$ git for-each-ref | wc -l
2196
```
Number of local branches:
```
$ git for-each-ref refs/heads | wc -l
18
```
Number of Remote branches:
```
$ git for-each-ref refs/remotes | wc -l
1346
```
Number of tags:
```
$ git for-each-ref refs/tags | wc -l
831
```

**Comment by DanPristupov** [comment]
@james-rabbit I'm testing the performance now and it's surprisingly fast. I use the Chrome repository (10 GB, 11577 references).

The Linux repository is huuge... It must freeze the UI too. Is that possible that you used it differently?

The only thing which does hang the UI is the changes view. Is that view always open in your case?

Just guessing: you always keep the changes view active and activating Fork causes refresh of changed file list which is very slow on huge a repository. When you opened Linux repo, you clicked around in the repository view but, not in "Changes" view.

If my theory is correct, I will update the Changes view to perform request on the background thread and update UI only if data changed.

**Comment by james-rabbit** [comment]
I just updated to Fork 1.0.56.1. 

There is still a significant delay when focusing the app and the "changes view" is active, but there is no delay when focusing on the app when it's in the "all commits" view on this version, like you guessed.

I failed to mention it before, but the same delay exists when I stage or unstage files in the changes view.

**Comment by james-rabbit** [comment]
Thanks! I'll report back after using it for a while. At first glance it is running much better though!

**Comment by james-rabbit** [comment]
After using it for a couple days, I think we can safely mark this issue as resolved when the next release is out! The UI is now very responsive, and the loading indicator at the top center makes it clear when some background git process is running.

**Comment by james-rabbit** [comment]
This is fixed as of 1.0.57.6, thank you!
